### PR TITLE
Handling null pointer in GetPartitionRequestDto

### PR DESCRIPTION
**Pull request by zhljen**

### CODE REVIEW

File: metacat-main/src/main/java/com/netflix/metacat/main/services/impl/PartitionServiceImpl.java [hunk 1]
```diff
@@ -137,12 +137,8 @@ public PartitionServiceImpl(
         final GetPartitionsRequestDto getPartitionsRequestDto
     ) {
 
-        String filterExpression = null;
-        List<String> partitionNames = null;
-        if (getPartitionsRequestDto != null) {
-            filterExpression = getPartitionsRequestDto.getFilter();
-            partitionNames = getPartitionsRequestDto.getPartitionNames();
-        }
+        final String filterExpression = getPartitionsRequestDto.getFilter();
```
Comment (ajoymajumdar) [inline metacat-main/src/main/java/com/netflix/metacat/main/services/impl/PartitionServiceImpl.java]: Why did we remove the null check?

Comment (zhljen) [reply metacat-main/src/main/java/com/netflix/metacat/main/services/impl/PartitionServiceImpl.java]: Agreed. will make the change.

File: metacat-main/src/main/java/com/netflix/metacat/main/services/impl/PartitionServiceImpl.java [hunk 2]
```diff
@@ -134,29 +135,28 @@ public PartitionServiceImpl(
...
+            converterUtil.toPartitionListRequest(getPartitionsRequestDto, pageable, sort);
 
-        String filterExpression = null;
-        List<String> partitionNames = null;
-        if (getPartitionsRequestDto != null) {
-            filterExpression = getPartitionsRequestDto.getFilter();
-            partitionNames = getPartitionsRequestDto.getPartitionNames();
-        }
+        final ConnectorRequestContext connectorRequestContext = converterUtil.toConnectorContext(metacatRequestContext);
+        final List<PartitionInfo> resultInfo = service
```
Comment (ajoymajumdar) [inline metacat-main/src/main/java/com/netflix/metacat/main/services/impl/PartitionServiceImpl.java]: This should come after the validation check that throws IllegalArugumentException.

Comment (zhljen) [reply metacat-main/src/main/java/com/netflix/metacat/main/services/impl/PartitionServiceImpl.java]: Sounds right.

**Pull request merged**